data/src/test/java/org/apache/iceberg/data/orc/TestGenericReadProjection.java

```diff
@@ -33,7 +33,8 @@ public class TestGenericReadProjection extends TestReadProjection {
   @Override
   protected Record writeAndRead(String desc, Schema writeSchema, Schema readSchema, Record record)
       throws IOException {
-    File file = temp.newFile(desc + ".orc");
+
+    File file = File.createTempFile("junit", null, temp);
```

Comment:
```suggestion
    File file = File.createTempFile(desc, ".orc", temp);
```

we want to keep the `.orc` suffix here. Also please remove the unnecessary newline that was added right above this line